.github/workflows/actions/download-archive/action.yml

```diff
@@ -16,5 +16,5 @@ runs:
         path: ${{ inputs.path }}
 
     - name: Extract Archive
-      run: unzip -q -o ${{ inputs.path }}/${{ inputs.filename }}
+      run: unzip -q -o ${{ inputs.path }}/${{ inputs.filename }} -d ${{ inputs.path }}
```

Comment:
This is kind of funky. We were always using the 'path' for the location of the tarball, rather than the location of where to extract things to!